crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -391,16 +394,43 @@ impl<'db> Class<'db> {
                     );
                 }
                 ClassBase::Class(class) => {
-                    if let member @ SymbolAndQualifiers(Symbol::Type(_, _), _) =
+                    if let member @ SymbolAndQualifiers(Symbol::Type(ty, boundness), qualifiers) =
                         class.own_instance_member(db, name)
                     {
-                        return member;
+                        // TODO: We could raise a diagnostic here if there are conflicting type qualifiers
```

Comment:
Probably the diagnostic would be raised by an eager Liskov check at the inheritance site, rather than here? But it doesn't hurt to have a TODO here until we do that.